src/utils/localisation/exportLanguages.ts

```diff
@@ -0,0 +1,99 @@
+import { getRequest } from '../helpers/fetchMethods'
+import config from '../../config'
+import { LanguageOption, LanguageStrings } from '../../contexts/Localisation'
+import defaultLanguageStrings from '../../utils/defaultLanguageStrings'
+import { mapValues } from 'lodash'
+import Papa from 'papaparse'
+import { DateTime } from 'luxon'
+
+export interface LanguageObject extends LanguageOption {
+  translations: LanguageStrings
+}
+
+export const exportLanguages = async (
+  languageOptions: LanguageOption[]
+): Promise<{ success: boolean; message: string }> => {
+  // Fetch all languages
+  const allLanguageStrings: { [key: string]: LanguageStrings } = await getRequest(
+    `${config.serverREST}/admin/all-languages`
+  )
+
+  // Combine language options with their translations
+  const languageObject: LanguageObject[] = languageOptions.map((language) => ({
+    ...language,
+    translations: allLanguageStrings[language.code],
+  }))
+
+  // Iterate over default strings and extract all translations
+  const translationsObject = mapValues(defaultLanguageStrings, (value, key) =>
+    getTranslations(value, key as keyof LanguageStrings, languageObject)
+  )
+
+  // Assemble orphans
+  const orphanKeys = languageObject.map((lang) => Object.keys(lang.translations)).flat()
+  const orphansObject = Object.fromEntries(
+    orphanKeys.map((key) => [
+      key,
+      getTranslations('', key as keyof LanguageStrings, languageObject),
+    ])
+  )
+
+  // Build CSV Rows and Columns
+  const row1Name = [
+    'Language:',
+    'System Default',
+    ...languageOptions.map((opt) => opt.languageName),
+  ]
+  const row2Description = [
+    'Description:',
+    'Internal values from which all others are translated',
+    ...languageOptions.map((opt) => opt.description),
+  ]
```

Comment:
Should this be localised too?

Reply:
I dunno -- these go onto the CSV output and I don't know if we want to worry about making the spreadsheet localised just yet. Especially since anyone who uses it would probably need to be an English speaker to make sense of the default strings.

I was kind of imagining that we might have a shared Google sheet for all the different languages which people can keep updated with their own language, in which case this wouldn't be useful to localise. But that's a discussion fo another day -- can we leave these for now?

Reply:
Yep. No problem. I liked you idea of the spreadsheet with each localisation, although I understood as each project would be using theirs. Also because the sheet is currently storing all the languages that are part of a project.